examples/07-Circuit/Virtual_Compliance.py

```diff
@@ -0,0 +1,258 @@
+"""
+Circuit: PCIE Virtual Compliance
+--------------------------------
+This example shows how you can generate a compliance report in PyAEDT using
+the new class Virtual Compliance.
+"""
+
+###############################################################################
+# Perform required imports
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+# Perform required imports and set paths.
+
+import os.path
+import pyaedt
+from pyaedt.generic.compliance import VirtualCompliance
+
+###############################################################################
+# Download example files
+# ~~~~~~~~~~~~~~~~~~~~~~
+# Download project and files needed to run the example.
+workdir = pyaedt.downloads.download_file('pcie_compliance')
+
+#workdir = r'C:\ansysdev\Models\Compliance\pcie\wf_pcie'
+projectdir = os.path.join(workdir, "project")
+
+
+###############################################################################
+# Launch Electronics Desktop
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Launch Electronics Desktop.
+
+d = pyaedt.Desktop(241, new_desktop_session=False, non_graphical=True)
+
+
+###############################################################################
+# Open and Solve Layout
+# ~~~~~~~~~~~~~~~~~~~~~
+# Open Hfss 3D Layout project and analyze it using Siwave solver.
+# Before solving we make sure that the model is solved from DC to 70GHz and that
+# causality and passivity are enforced.
+
+h3d = pyaedt.Hfss3dLayout(os.path.join(projectdir, "PCIE_GEN5_only_layout.aedtz"), specified_version=241)
+h3d.remove_all_unused_definitions()
+h3d.edit_cosim_options(simulate_missing_solution=False)
+h3d.setups[0].sweeps[0].props["EnforcePassivity"] = True
+h3d.setups[0].sweeps[0].props["Sweeps"]["Data"] =  'LIN 0MHz 70GHz 0.1GHz'
+h3d.setups[0].sweeps[0].props["EnforceCausality"] = True
+h3d.setups[0].sweeps[0].update()
+h3d.analyze()
+touchstone_path = h3d.export_touchstone()
+
+###############################################################################
+# Create LNA project
+# ~~~~~~~~~~~~~~~~~~
+# LNA Setup is useful to retrieve touchstone files
+# and generate frequency domain reports.
+
+cir = pyaedt.Circuit(projectname=h3d.project_name, designname="Touchstone")
+
+###############################################################################
+# Add dynamic link object
+# ~~~~~~~~~~~~~~~~~~~~~~~
+# The layout project will be added as a dynamic link object.
+# It means that in case of modifications,
+# the circuit will always be linked to updated version of results.
```

Comment:
```suggestion
# Add the layout project as a dynamic link object
# so that if modifications are made, the circuit
# is always linked to the updated version of the results.
```